File: common/src/main/java/juuxel/adorn/block/BlockWithDescription.java
package juuxel.adorn.block;

import juuxel.adorn.component.AdornComponentTypes;
import juuxel.adorn.component.ItemDescription;
import net.minecraft.world.level.block.Block;

/**
 * Can be added to a block to provide a description for the block item
 * when registered using {@link juuxel.adorn.lib.registry.RegistryHelper#registerBlock}.
 */
public interface BlockWithDescription extends BlockWithItemComponents {
    default String getDescriptionKey() {
        return ((Block) this).getDescriptionId() + ".description";
    }

    @Override
    default void addItemComponents(ComponentConsumer consumer) {
        consumer.add(AdornComponentTypes.DESCRIPTION, ItemDescription.ofTranslation(getDescriptionKey()));
    }
}

File: common/src/main/kotlin/juuxel/adorn/block/BlockWithDescription.kt
package juuxel.adorn.block

import net.minecraft.block.Block

/**
 * Can be added to a block to provide a description for the block item
 * when registered using [juuxel.adorn.lib.RegistryHelper.registerBlock].
 */
interface BlockWithDescription {
    val descriptionKey: String get() = (this as Block).translationKey + ".description"
}
